src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -754,7 +754,7 @@ public void finishSessionInit(ServerCnxn cnxn, boolean valid) {
             cnxn.sendBuffer(bb);
 
             if (valid) {
-                LOG.info("Established session 0x"
+                LOG.debug("Established session 0x"
```

Comment:
This sounds pretty useful INFO level log. If we turn it into DEBUG level, are there any other equivalent INFO level logging where we can derive the session establishment details? If not, probably better to leave this as INFO?

Reply:
I'd leave this one as INFO too.

Reply:
i think this should go to debug. if you have thousands of clients that are reconnecting after a leader change the log fills with these. if you are really interested in this line you can add a log4j config that will show it. but this really creates a lot of noise in the log.

Reply:
The attributes of an established connection don't appear in other INFO logs but remain accessible through jmx and through ServerCnxn::dumpConnectionInfo via 'cons'.

I'd agree with Ben that this is makes parsing the logs hard when the number of client connections is high and that it's better to isolate it via log level so it can be turned off in that scenario or turned on (via debug) if needed for low client-supporting ensembles.